_posts/2019-08-29-wfs3.md

```diff
@@ -0,0 +1,503 @@
+---
+layout: left
+title: QGIS Server and WFS3
+active: Articles
+---
+
+# QGIS Server and WFS3
+
+<hr>
+{:.post-icon}
+![date]({{ site.baseurl }}/images/posts/calendar.png){:width="18px"} &emsp; *August 25, 2019*
+
+{:.post-icon}
+![tags]({{ site.baseurl }}/images/posts/tag.png){:width="18px"} &emsp; *QGIS Server, OGC, WFS3*
+
+{:.post-icon}
+![avatar]({{ site.baseurl }}/images/posts/avatar.png){:width="18px"} &emsp; *Paul Blottiere and Alessandro Pasotti*
+<hr>
+
+
+## Introduction
+
+There are various map servers available in the Open Source world, each with
+advantages and disadvantages. <a href="https://mapserver.org/">Map Server</a>
+and <a href="http://geoserver.org/">GeoServer</a> are the most famous, but QGIS
+provides a map server too, named <a href="https://docs.qgis.org/3.4/en/docs/user_manual/working_with_ogc/server/index.html">QGIS Server</a>.
+One of the main advantages of QGIS Server is the possibility to directly
+configure the project to serve through QGIS Desktop. Indeed, QGIS Desktop acts
+like a WYSIWYG tool because the rendering engine used by QGIS Server is exactly
+the same.
+
+QGIS Server provides some classical OGC services like WMS, WFS, WCS or even
+WMTS since QGIS 3.4. Furthermore, the certification process has been reached for WMS
+1.3.0 since QGIS 3.0 and has also been renewed for the LTR version 3.4.
+
+<br/>
+![OGC Badge]({{ site.baseurl }}/images/posts/wfs3/wms_ogc_badge.png){:class="center-image" width="175px"}
+<br/>
+
+And from now on, thanks to the work of
+<a href="https://github.com/elpaso">Alessandro Pasotti</a>
+in the light of the discussion held on the dedicated
+<a href="https://github.com/qgis/QGIS-Enhancement-Proposals/issues/144">QEP</a>,
+QGIS Server 3.10 is going to provide WFS 3 support. By the way, an experimental
+driver already exists for
+<a href="https://gdal.org/drivers/vector/wfs3.html">GDAL</a> and a prototype has
+been implemented in
+<a href="https://www.geo-solutions.it/blog/wfs3-geoserver/">GeoServer</a>.
+
+Let's take a look to the possibilities given by this
+brand-new protocol.
+<hr>
+
+
+## Quick reminder about WFS 1.X
+
+WFS, or Web Feature Service, is an Open Geospatial Consortium protocol aiming
+at manipulating vector features (lines, points, polygons, ...).
+
+To interact with a map server implementing WFS 1.X, a HTTP request has to be
+sent with particular parameters like a `NAME`{:.inline} for indicating a
+specific layer to work on or a `BBOX`{:.inline} to specify an extent.  Of
+course, there are numerous parameters, and some map servers, like QGIS Server,
+implement custom parameters (meaning parameters not described by the OGC
```

Comment:
maybe use "vendor" instead of "custom", but I don't really know if the readers know what "vendor" means.

Reply:
Actually, I've had the same question... But as you mentioned, I think that "custom" is more explicit than "vendor" for most of the potential readers.